src/Data/Product/Properties.agda

```diff
@@ -23,7 +25,17 @@ module _ {a b} {A : Set a} {B : A → Set b} where
   ,-injectiveˡ : ∀ {a c} {b : B a} {d : B c} → (a , b) ≡ (c , d) → a ≡ c
   ,-injectiveˡ refl = refl
 
-  -- See also Data.Product.Properties.WithK.,-injectiveʳ.
+  ,-injective≡ʳ : ∀ {a b} {c : B a} {d : B b} → UIP A → (a , c) ≡ (b , d) → (q : a ≡ b) → subst B q c ≡ d
+  ,-injective≡ʳ {c = c} {.c} u refl q = cong (λ x → subst B x c) (u q refl)
+
+  ,-injectiveUIPʳ : ∀ {a} {b c : B a} → UIP A → (Σ A B ∋ (a , b)) ≡ (a , c) → b ≡ c
+  ,-injectiveUIPʳ u p = ,-injective≡ʳ u p refl
+
+  ≡-dec : Decidable {A = A} _≡_ → (∀ {a} → Decidable {A = B a} _≡_) →
```

Comment:
Probably worth using this opportunity to use [`DecidableEquality`](http://agda.github.io/agda-stdlib/Relation.Binary.Definitions.html#4883)
rather than the `Decidable {A = ?} _≡_` pattern.

Reply:
Sure